src/main/java/ome/files/performance/OMETIFFPerformance.java

```diff
@@ -0,0 +1,201 @@
+import loci.common.services.ServiceFactory;
+
+import loci.formats.meta.MetadataStore;
+import loci.formats.meta.MetadataRetrieve;
+import loci.formats.ome.OMEXMLMetadata;
+import loci.formats.ome.OMEXMLMetadataImpl;
+import loci.formats.services.OMEXMLService;
+import loci.formats.services.OMEXMLServiceImpl;
+import loci.formats.in.OMETiffReader;
+import loci.formats.out.OMETiffWriter;
+import loci.formats.out.TiffWriter;
+import loci.formats.FormatReader;
+import loci.formats.FormatWriter;
+
+import loci.formats.tiff.IFD;
+import loci.formats.tiff.TiffParser;
+
+public final class OMETIFFPerformance {
+
+  public static void main(String[] args) {
+    if (args.length != 4)
+      {
+        System.out.println("Usage: MetadataPerformance iterations inputfile outputfile resultfile\n");
+        System.exit(1);
+      }
+
+    Result result = null;
+
+    try {
+      int iterations = Integer.parseInt(args[0]);
+      Path infile = Paths.get(args[1]);
+      Path outfile = Paths.get(args[2]);
+      Path resultfile = Paths.get(args[3]);
+
+      ServiceFactory factory = new ServiceFactory();
+      OMEXMLService service = factory.getInstance(OMEXMLService.class);
+
+      result = new Result(resultfile);
+
+      for(int i = 0; i < iterations; i++) {
+        OMEXMLMetadata meta = new OMEXMLMetadataImpl();
+        List<List<byte[]>> pixels = new ArrayList();
+
+        Timepoint read_start = new Timepoint();
+        Timepoint read_init = null;
+
+        {
+          System.out.print("pass " + i + ": read init...");
+          System.out.flush();
+          FormatReader reader = new OMETiffReader();
+          reader.setMetadataStore(meta);
+          reader.setId(infile.toString());
+          System.out.println("done");
+          System.out.flush();
+
+          read_init = new Timepoint();
+
+          for (int series = 0; series <reader.getSeriesCount(); ++series) {
+            System.out.print("pass " + i + ": read series " + series + ": ");
+            System.out.flush();
+            reader.setSeries(series);
+
+            List<byte[]> planes = new ArrayList();
+            pixels.add(planes);
+
+            for (int plane = 0; plane < reader.getImageCount(); ++plane) {
+              byte[] data = reader.openBytes(plane);
+              planes.add(data);
+              System.out.print(".");
+              System.out.flush();
+            }
+            System.out.println("done");
+            System.out.flush();
+          }
+          reader.close();
+        }
+
+        Timepoint read_end = new Timepoint();
+
+        result.add("ometiffdata.read", infile, read_start, read_end);
+        result.add("ometiffdata.read.init", infile, read_start, read_init);
+        result.add("ometiffdata.read.pixels", infile, read_init, read_end);
+
+        Files.deleteIfExists(outfile);
+
+        Timepoint write_start = new Timepoint();
+        Timepoint write_init = null;
+        Timepoint close_start = null;
+
+          {
```

Comment:
Are the new braces here for readability? Also some of the indentations appear to be out of sync